src/contribute/ptransform-style-guide.md

```diff
@@ -452,39 +452,61 @@ public static class FooV2 {
 
 #### Validation
 
-* Validate individual parameters in `.withBlah()` methods. Error messages should mention the method being called, the actual value and the range of valid values.
-* Validate inter-parameter invariants in the `PTransform`'s `.validate()` method.
+* Validate individual parameters in `.withBlah()` methods using `checkArgument()`. Error messages should mention the name of the parameter, the actual value, and the range of valid values.
+* Validate parameter combinations and missing required parameters in the `PTransform`'s `.validate()` or `.expand()` method.
```

Comment:
Swap the order of these methods; failing in `expand` is generally preferable, because it is earlier in the lifecycle of a PTransform.